lib/chef/provisioning/fog_driver/driver.rb

```diff
@@ -424,46 +425,63 @@ def wait_for_transport(action_handler, machine_spec, machine_options, server)
       end
     end
 
+    # Check given IP already attached and attach if necessary
+    # TODO update to remove old floating IPs
     def attach_floating_ips(action_handler, machine_spec, machine_options, server)
-      # TODO this is not particularly idempotent. OK, it is not idempotent AT ALL.  Fix.
-      if option_for(machine_options, :floating_ip_pool)
+      pool = option_for(machine_options, :floating_ip_pool)
+      floating_ip = option_for(machine_options, :floating_ip)
+      if pool
         Chef::Log.info 'Attaching IP from pool'
-        action_handler.perform_action "attach floating IP from #{option_for(machine_options, :floating_ip_pool)} pool" do
-          attach_ip_from_pool(server, option_for(machine_options, :floating_ip_pool))
+        allocated_addresses = server.addresses[pool]
+        unless allocated_addresses.nil?
+          floating_ips = allocated_addresses.select {|a_info| a_info['OS-EXT-IPS:type']=='floating'}
+          if floating_ips.length>0
```

Comment:
You are right - I made this update.

Is it possible to be assigned an IP from the pool without it being floating:question: